### PR TITLE
Raise value error

### PR DESCRIPTION
**Pull request by zhaochenyang20**
# Description

https://github.com/neulab/prompt2model/issues/231

Another troublesome thing is that due to the release of the constraints of the packages' version, our new `transformers` now support MPS devices, but I can't find a way to debug it on my Mac.

https://github.com/neulab/prompt2model/issues/317

# References

- NA

# Blocked by

- NA
- (or link to PRs)

### CODE REVIEW

File: prompt2model/dataset_generator/openai_gpt.py [hunk 1]
```diff
@@ -99,12 +99,12 @@ def __init__(
...
         self.api_key: str | None = api_key if api_key else os.environ["OPENAI_API_KEY"]
-        assert self.api_key is not None and self.api_key != "", (
-            "API key must be provided"
-            + " or set the environment variable with `export OPENAI_API_KEY=<your key>`"
-        )
-        if max_api_calls:
-            assert max_api_calls > 0, "max_api_calls must be > 0"
+        if self.api_key is None or self.api_key == "":
+            raise ValueError(
+                "API key must be provided or set the environment variable with `export OPENAI_API_KEY=<your key>`."  # noqa E501
```
Comment (neubig) [inline prompt2model/dataset_generator/openai_gpt.py:104]:
```suggestion
                "API key must be provided or set the environment variable "
                "with `export OPENAI_API_KEY=<your key>`."
```

File: prompt2model/dataset_generator/openai_gpt.py [hunk 2]
```diff
@@ -321,7 +321,8 @@ def apply_multi_vote_to_construct_generated_dataset(
             Currently generated dataset with multi-vote filtering applied.
         """
         # Ensure that multi-vote filtering is enabled.
-        assert self.filter_duplicated_examples
+        if self.filter_duplicated_examples is False:
```
Comment (neubig) [inline prompt2model/dataset_generator/openai_gpt.py:324]:
```suggestion
        if not self.filter_duplicated_examples:
```

File: prompt2model/model_executor/generate.py [hunk 3]
```diff
@@ -32,14 +32,17 @@ def generate(
...
             "beam",  # beam search.
             "top_k",  # top_k sampling.
             "top_p",  # top_p sampling.
             "greedy",  # greedy search.
             "intersect",  # If both top_k and top_p are set, the model will
             # sample from the intersection of the top-k tokens and the top-p tokens.
-        ], "Only support top_k/top_p/intersect sampling and beam/greedy search for inference."  # noqa E501
+        ]:
+            raise ValueError(
+                "Only support top_k/top_p/intersect sampling and beam/greedy search for inference."  # noqa E501
```
Comment (neubig) [inline prompt2model/model_executor/generate.py:44]:
```suggestion
                "Only top_k/top_p/intersect sampling and beam/greedy "
                "search are supported for inference."
```

File: prompt2model/model_executor/generate.py [hunk 4]
```diff
@@ -190,7 +193,10 @@ def make_single_prediction(
         expected_num_examples = 1
         inference_dataset = datasets.Dataset.from_dict({"model_input": [model_input]})
         inference_column = "model_input"
-        assert len(inference_dataset) == expected_num_examples
+        if len(inference_dataset) != expected_num_examples:
+            raise ValueError(
+                f"Expected {expected_num_examples} examples, but got {len(inference_dataset)}."  # noqa E501
```
Comment (neubig) [inline prompt2model/model_executor/generate.py:198]:
```suggestion
                f"Expected {expected_num_examples} examples, "
                f"but got {len(inference_dataset)}."
```

File: prompt2model/model_retriever/description_based_retriever.py [hunk 5]
```diff
@@ -105,16 +105,18 @@ def __init__(
         self.load_model_info()
 
         if self.use_bm25:
-            assert (
-                search_index_path is None
-            ), "BM25 expects a search index path with a particular format, so search_index_path should not be provided."  # noqa E501
+            if search_index_path is not None:
+                raise ValueError(
+                    "BM25 expects a search index path with a particular format, so search_index_path should not be provided."  # noqa E501
```
Comment (neubig) [inline prompt2model/model_retriever/description_based_retriever.py:110]:
```suggestion
                    "BM25 expects a search index path with a particular format, "
                    "so search_index_path should not be provided."
```

File: prompt2model/model_retriever/description_based_retriever.py [hunk 6]
```diff
@@ -105,16 +105,18 @@ def __init__(
...
             self.bm25_index_name = bm25_index_name
             self._search_index_path = retriv.paths.index_path(self.bm25_index_name)
         else:
             if search_index_path is not None:
-                assert not os.path.isdir(
-                    search_index_path
-                ), f"Search index must either be a valid file or not exist yet. But {search_index_path} is provided."  # noqa E501
+                if os.path.isdir(search_index_path):
+                    raise ValueError(
+                        f"Search index must either be a valid file or not exist yet. But {search_index_path} is provided."  # noqa E501
```
Comment (neubig) [inline prompt2model/model_retriever/description_based_retriever.py:118]:
```suggestion
                        f"Search index must either be a valid file or not exist yet. "
                        f"But {search_index_path} is provided."
```

File: prompt2model/model_retriever/description_based_retriever.py [hunk 7]
```diff
@@ -204,7 +206,8 @@ def scale_similarity_score(
 
     def bm25_index_exists(self):
         """Check if a BM25 index exists."""
-        assert self.use_bm25 is True
+        if self.use_bm25 is False:
```
Comment (neubig) [inline prompt2model/model_retriever/description_based_retriever.py:209]:
```suggestion
        if not self.use_bm25:
```

File: prompt2model/prompt_parser/instr_parser.py [hunk 8]
```diff
@@ -46,12 +46,12 @@ def __init__(
...
         self.api_key: str | None = api_key if api_key else os.environ["OPENAI_API_KEY"]
-        assert self.api_key is not None and self.api_key != "", (
-            "API key must be provided"
-            + " or set the environment variable with `export OPENAI_API_KEY=<your key>`"
-        )
-        if max_api_calls:
-            assert max_api_calls > 0, "max_api_calls must be > 0"
+        if self.api_key is None or self.api_key == "":
+            raise ValueError(
+                "API key must be provided or set the environment variable with `export OPENAI_API_KEY=<your key>`."  # noqa E501
```
Comment (neubig) [inline prompt2model/prompt_parser/instr_parser.py:51]:
```suggestion
                "API key must be provided or set the environment variable with "
                "`export OPENAI_API_KEY=<your key>`."
```

File: prompt2model/utils/openai_tools.py [hunk 9]
```diff
@@ -47,10 +47,10 @@ def __init__(self, api_key: str | None, model_name: str = "gpt-3.5-turbo"):
             model_name: Name fo the OpenAI model to use (by default, gpt-3.5-turbo).
         """
         openai.api_key = api_key if api_key else os.environ["OPENAI_API_KEY"]
-        assert openai.api_key is not None and openai.api_key != "", (
-            "API key must be provided"
-            + " or set the environment variable with `export OPENAI_API_KEY=<your key>`"
-        )
+        if openai.api_key is None or openai.api_key == "":
+            raise ValueError(
+                "API key must be provided or set the environment variable with `export OPENAI_API_KEY=<your key>`."  # noqa E501
```
Comment (neubig) [inline prompt2model/utils/openai_tools.py:52]:
```suggestion
                "API key must be provided or set the environment variable "
                "with `export OPENAI_API_KEY=<your key>`."
```

**Review by neubig — APPROVED** [review]
Looks great! I left several suggestions about how to remove `noqa`s by splitting strings into multiple lines (we should probably avoid introducing noqas unless really necessary), but I'm happy to merge after those are incorporated.

**Comment by neubig** [comment]
Oh, woops. I didn't realize that auto-merge was turned on. I guess I shouldn't have clicked "approve". It's OK though, they could probably be fixed later.

**Comment by zhaochenyang20** [comment]
Sure! I gonna fix it right not.

**Pull request merged**